lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -3050,7 +3060,7 @@ LogicalResult ICmpOp::canonicalize(ICmpOp op, PatternRewriter &rewriter) {
                                            : APInt::getZero(width));
           replaceOpWithNewOpAndCopyName<ICmpOp>(rewriter, op, op.getPredicate(),
                                                 replicateOp.getInput(), cst,
-                                                false);
+                                                op.getTwoState());
```

Comment:
Can you split these two state preservations into a separate PR?

Reply:
Sure, https://github.com/llvm/circt/pull/4695 is the PR.